packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -507,26 +509,32 @@ export class StateMachine extends StateMachineBase {
   }
 
   private buildLoggingConfiguration(logOptions: LogOptions): CfnStateMachine.LoggingConfigurationProperty {
-    // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
-    this.addToRolePolicy(new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: [
-        'logs:CreateLogDelivery',
-        'logs:GetLogDelivery',
-        'logs:UpdateLogDelivery',
-        'logs:DeleteLogDelivery',
-        'logs:ListLogDeliveries',
-        'logs:PutResourcePolicy',
-        'logs:DescribeResourcePolicies',
-        'logs:DescribeLogGroups',
-      ],
-      resources: ['*'],
-    }));
+    if (logOptions.level !== LogLevel.OFF && !logOptions.destination) {
+      throw new Error('Logs destination is required when level is not OFF.');
+    }
```

Comment:
shall this validation be done earlier (at line [427](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts#L427)) with other validations instead?
```
if (!logs && !logs.level != LogLevel.OFF && !logs.destination) { ...
```

reasons for ask: 
- with current change, validations will be split across various places and as this file is big (and keep getting bigger), hence, it will become difficult to have complete picture of validation 
- ideally, caller shall not even call `buildLoggingConfiguration` if prerequisites are not met

Reply:
added a separate validation method validateLogOptions()